packages/flutter/lib/src/widgets/layout_builder.dart

```diff
@@ -0,0 +1,192 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'debug.dart';
+import 'framework.dart';
+
+import 'package:flutter/rendering.dart';
+
+/// The signature of the [LayoutBuilder] builder function.
+typedef Widget LayoutWidgetBuilder(BuildContext context, Size size);
+
+/// Builds a widget tree that can depend on the parent widget's size.
+///
+/// Similar to the [Builder] widget except that the framework calls the [builder]
+/// function at layout time and provides the parent widget's size. This is useful
+/// when the parent constrains the child's size and doesn't depend on the child's
+/// intrinsic size.
+class LayoutBuilder extends RenderObjectWidget {
+  LayoutBuilder({ Key key, this.builder }) : super(key: key);
+
+  /// Called at layout time to construct the widget tree. The builder must not
+  /// return null.
```

Comment:
I think we should either allow it to return null, or require that the callback itself be non-null. It's weird to allow it to be null but not allow it to return null.